### PR TITLE
Documentation for OpenShift deployment

### PR DESCRIPTION
**Pull request by mirroredge**
Updated the docs to show how to deploy into OpenShift

## Description
I mirrored the k8s deployment instructions and modified them to work with OpenShift. The changes were minimal mostly pointing to the correct documentation and using oc commands.

## Motivation and Context
https://github.com/openfaas/faas-netes/issues/349
- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
I ran through the instructions I created. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`

### CODE REVIEW

**Comment by alexellis** [comment]
Hi @mirroredge thanks for this PR 💯 

I have only just seen it, but noticed a couple of things we should amend:

* squash down the commits
* rebase as per the latest changes
* sign off that commit, because we have one or a merge commit which has no "Signed-off-by" message

I would also recommend removing duplication between the K8s guide and the OpenShift guide where possible. Do you have any ideas on how we could do that?

@LucasRoesler should we generate a `yaml_openshift` folder? This would be done by helm as we are currently generating `yaml`, but pre-package open shift extensions like "routes".

File: docs/deployment/openshift.md [hunk 1]
```diff
@@ -1,4 +1,286 @@
...
+    ```bash
+    $ oc new-project openfaas
+    $ oc new-project openfaas-fn
+    ```
+
+    Now deploy OpenFaaS:
+
+    ```bash
+    $ cd faas-netes && \
+    oc apply -R -f ./openshift
```
Comment (LucasRoesler) [inline docs/deployment/openshift.md]: Where does this folder come from? Is there a PR in faas-netes to add this?  I looked but perhaps I missed it.

**Comment by LucasRoesler** [comment]
@alexellis i think it makes sense to either generate or at the minimum give instructions for how to generate that folder like we do with yaml.  

If we generate it, should we consider putting the yaml folders in a `deployment` directory, to keep things tidy?

File: docs/deployment/openshift.md (hunk 1 above)
Comment (mirroredge) [reply docs/deployment/openshift.md]: Correct. Its in this PR: https://github.com/openfaas/faas-netes/pull/407

**Pull request closed without merging**